pkg/api/v1/types.go

```diff
@@ -2503,6 +2503,8 @@ type EventSource struct {
 	Component string `json:"component,omitempty"`
 	// Host name on which the event is generated.
 	Host string `json:"host,omitempty"`
+	// Scheduler name on which the event is generated.
+	SchedulerName string `json:"schedulerName,omitempty"`
```

Comment:
I don't think it makes sense to have a schedulerName field here. I think the scheduler name is actually part of the component field (e.g., "scheduler-batch"). If we really want to separate it, I think we should define something like shardName, instanceName, role, etc. Events are more general than scheduler and can't reference schedulers specifically.

Alternatively, we could add a "componentClass" field, set *that* to scheduler, and set the current component field to the name of the scheduler. "shard" may actually be a different thing entirely, like I can imagine wanting to say scheduler, batch, shard 3.

@davidopp may have opinions too. The only thing I feel strongly about is that we can't reference specific components like schedulers in this object.